compiler/codegen/cg-expr.cpp

```diff
@@ -4886,6 +4887,41 @@ DEFINE_PRIM(GPU_BLOCKDIM_Z)  { ret = codegenCallToPtxTgtIntrinsic("llvm.nvvm.rea
 DEFINE_PRIM(GPU_GRIDDIM_X)   { ret = codegenCallToPtxTgtIntrinsic("llvm.nvvm.read.ptx.sreg.nctaid.x"); }
 DEFINE_PRIM(GPU_GRIDDIM_Y)   { ret = codegenCallToPtxTgtIntrinsic("llvm.nvvm.read.ptx.sreg.nctaid.y"); }
 DEFINE_PRIM(GPU_GRIDDIM_Z)   { ret = codegenCallToPtxTgtIntrinsic("llvm.nvvm.read.ptx.sreg.nctaid.z"); }
+
+DEFINE_PRIM(GPU_ALLOC_SHARED) {
+#ifdef HAVE_LLVM
```

Comment:
I don't know how the `DEFINE_PRIM` macro works, but is this a problem if you don't have LLVM? I guess it can be if you use the primitive without LLVM, but that's on you :)

But does it cause any other issues? Our CI must be checking against that, but I'd feel more comfrotable if you can do `make` / `make check` in an LLVM=none environment.

Reply:
> I don't know how the DEFINE_PRIM macro works, but is this a problem if you don't have LLVM?

Yes you'll get compile-time errors complaining that it can't find the llvm-related symbols.

> I guess it can be if you use the primitive without LLVM, but that's on you :)

True although maybe we should assert; although I suppose we would want to make it possible for the user to write (Chapel code) like this:

```chpl
  if compilingForGpu() {
    __primitive("gpu ...")
  } else {
    // some alternate code pass
  }
```

In which case maybe the primitives should emit some kind of runtime warning. I'm not sure how aggressive we want to be about informing the user that there's doing something silly though.

> Our CI must be checking against that, but I'd feel more comfrotable if you can do make / make check in an LLVM=none environment.

I believe it will (I haven't tested it this time but I do vaguely remember this failing with some other PR I did).  Anyway, I've tried compiling it with `CHPL_LLVM=none` on my mac and it seems to work fine.

Reply:
> True although maybe we should assert; although I suppose we would want to make it possible for the user to write (Chapel code) like this:

In that code, I would expect `compilingForGpu()` to return a param, and that conditional to be param-folded much earlier in compilation. So that primitive would not be codegenned if compiling without GPU.

But in any case, I don't think this needs any action at the moment.